### PR TITLE
Add easy loading of extensions

### PR DESCRIPTION
**Pull request by jdevalk**
⚠️ Not entirely happy with how I made the filter backwards compatible here, I'm open to suggestions :) ⚠️ 

Added these features:

enhanced-measurements

Notable other changes:
* Remove the requirement for YODA conditions form WPCS, as they're stupid. Discussion about removing this from WPCS itself [is ongoing](https://make.wordpress.org/core/2022/06/14/upcoming-disallow-assignments-in-conditions-and-remove-the-yoda-condition-requirement-for-php/).
* Fixed the accessibility of checkboxes and text input fields by giving them proper associated labels.
* Slightly changed how the analytics link worked, as [can be seen here](#diff-8ebf0dcfabde08c55f7fec8dcbce2797f44c10d40f159b882a6af5e87821e505R119-R121)

### CODE REVIEW

**Comment by jdevalk** [comment]
@mehul0810 @ukutaht I've merged `develop` back into this and fixed the warnings. I think this is now ready for review.

File: src/Admin/Settings/API.php [hunk 1]
```diff
@@ -96,38 +96,37 @@ public function render_text_field( array $field ) {
...
 		$field_value = ! empty( $settings[ $group['slug'] ] ) ? $settings[ $group['slug'] ] : false;
 		$is_checked  = checked( $field_value, true, false );
-		$domain_name = ! empty( $settings['domain_name'] ) ? $settings['domain_name'] : '';
 		ob_start();
 		?>
 		<div class="plausible-analytics-admin-field">
 			<div class="plausible-analytics-admin-field-header">
 				<label for="">
-					<?php esc_html_e( $group['label'], 'plausible-analytics' ); ?>
+					<?php echo $group['label']; ?>
```
Comment (jdevalk) [inline src/Admin/Settings/API.php:112]: These are translated elsewhere, you can't pass variables into translation functions in WP.

File: src/Admin/Settings/API.php [hunk 2]
```diff
@@ -96,38 +96,37 @@ public function render_text_field( array $field ) {
...
 				</label>
-				<?php if ( $toggle ) { ?>
+				<?php if ( $toggle === true ) { ?>
 				<label class="plausible-analytics-switch">
-					<input <?php echo $is_checked; ?> class="plausible-analytics-switch-checkbox" name="plausible_analytics_settings[<?php esc_attr_e( $group['slug'], 'plausible-analytics' ); ?>]" value="1" type="checkbox">
+					<input <?php echo $is_checked; ?> class="plausible-analytics-switch-checkbox" name="plausible_analytics_settings[<?php echo $group['slug']; ?>]" value="1" type="checkbox">
```
Comment (jdevalk) [inline src/Admin/Settings/API.php:116]: Same as above.

File: src/Admin/Settings/API.php [hunk 3]
```diff
@@ -96,38 +96,37 @@ public function render_text_field( array $field ) {
...
 					<span class="plausible-analytics-switch-slider"></span>
 				</label>
-				<?php } else { ?>
-					<a target="_blank" class="plausible-analytics-link" href="<?php echo esc_url( "https://plausible.io/{$domain_name}" ); ?>">
-						<?php esc_html_e( 'Open Analytics', 'plausible-analytics' ); ?>
+				<?php } elseif ( is_array( $toggle ) ) { ?>
+					<a target="_blank" class="plausible-analytics-link" href="<?php echo $toggle['anchor']; ?>">
+						<?php echo $toggle['label']; ?>
```
Comment (jdevalk) [inline src/Admin/Settings/API.php:121]: This allows either linking to analyics, showing a toggle, or doing nothing.

File: src/Admin/Settings/API.php [hunk 4]
```diff
@@ -96,38 +96,37 @@ public function render_text_field( array $field ) {
...
 					</a>
 				<?php } ?>
 			</div>
 			<div class="plausible-analytics-admin-field-body">
 				<?php
 				if ( ! empty( $fields ) ) {
 					foreach ( $fields as $field ) {
-						echo call_user_func( [ $this, "render_{$field['type']}_field" ], $field );
+						echo call_user_func( [ $this, "render_{$field['type']}_field" ], $field ) . '<br/>';
```
Comment (jdevalk) [inline src/Admin/Settings/API.php:129]: Added a line break after each field here, for readability.

File: src/Includes/Helpers.php [hunk 5]
```diff
@@ -113,7 +118,15 @@ public static function display_toggle_switch( $name ) {
...
-		return get_option( 'plausible_analytics_settings', [] );
+		$settings = get_option( 'plausible_analytics_settings', [] );
+
+		// Keep around for backwards compatibility reasons.
+		$track_outbound_links = apply_filters( 'plausible_analytics_enable_outbound_links', isset( $settings['outbound-links'][0] ) ? $settings['outbound-links'][0] : true );
+		if ( $track_outbound_links ) {
+			$settings['outbound-links'][0] = 1;
+		}
+
+		return $settings;
```
Comment (jdevalk) [inline src/Includes/Helpers.php:129]: I'm not 100% happy with this bit of code, as it's... not ideally functional. Would love to hear other suggestions.

**Comment by mehul0810** [comment]
@jdevalk There are some more merge conflicts. Can you please take care of those as well?

**Comment by jdevalk** [comment]
@mehul0810 done

**Review by mehul0810 — APPROVED** [review]

**Pull request merged**